src/app/clusters/commodity-tariff-server/commodity-tariff-server.cpp

```diff
@@ -0,0 +1,172 @@
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "commodity-tariff-server.h"
+
+#include <app/AttributeAccessInterface.h>
+#include <app/AttributeAccessInterfaceRegistry.h>
+#include <app/CommandHandlerInterfaceRegistry.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/InteractionModelEngine.h>
+#include <app/util/attribute-storage.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::CommodityTariff;
+using namespace chip::app::Clusters::CommodityTariff::Attributes;
+
+using chip::Protocols::InteractionModel::Status;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace CommodityTariff {
+
+CHIP_ERROR Instance::Init()
+{
+    ReturnErrorOnFailure(CommandHandlerInterfaceRegistry::Instance().RegisterCommandHandler(this));
+    VerifyOrReturnError(AttributeAccessInterfaceRegistry::Instance().Register(this), CHIP_ERROR_INCORRECT_STATE);
+
+    return CHIP_NO_ERROR;
+}
+
+void Instance::Shutdown()
+{
+    CommandHandlerInterfaceRegistry::Instance().UnregisterCommandHandler(this);
+    AttributeAccessInterfaceRegistry::Instance().Unregister(this);
+}
+
+bool Instance::HasFeature(Feature aFeature) const
+{
+    return mFeature.Has(aFeature);
+}
+
+// AttributeAccessInterface
+CHIP_ERROR Instance::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    switch (aPath.mAttributeId)
+    {
+    case TariffInfo::Id:
+        return aEncoder.Encode(mDelegate.GetTariffInfo());
+    case TariffUnit::Id:
+        return aEncoder.Encode(mDelegate.GetTariffUnit());
+    case StartDate::Id:
+        return aEncoder.Encode(mDelegate.GetStartDate());
+    case DayEntries::Id:
+        return aEncoder.Encode(mDelegate.GetDayEntries());
+    case DayPatterns::Id:
+        return aEncoder.Encode(mDelegate.GetDayPatterns());
+    case CalendarPeriods::Id:
+        return aEncoder.Encode(mDelegate.GetCalendarPeriods());
+    case IndividualDays::Id:
+        return aEncoder.Encode(mDelegate.GetIndividualDays());
+    case CurrentDay::Id:
+        return aEncoder.Encode(mDelegate.GetCurrentDay());
+    case NextDay::Id:
+        return aEncoder.Encode(mDelegate.GetNextDay());
+    case CurrentDayEntry::Id:
+        return aEncoder.Encode(mDelegate.GetCurrentDayEntry());
+    case CurrentDayEntryDate::Id:
+        return aEncoder.Encode(mDelegate.GetCurrentDayEntryDate());
+    case NextDayEntry::Id:
+        return aEncoder.Encode(mDelegate.GetNextDayEntry());
+    case NextDayEntryDate::Id:
+        return aEncoder.Encode(mDelegate.GetNextDayEntryDate());
+    case TariffComponents::Id:
+        return aEncoder.Encode(mDelegate.GetTariffComponents());
+    case TariffPeriods::Id:
+        return aEncoder.Encode(mDelegate.GetTariffPeriods());
```

Comment:
This is not going to allow chunking of those lists.  That's probably not what's intended.  That applies to all the list-typed attributes.